modules/components/social/org.wso2.carbon.social.sql/src/main/java/org/wso2/carbon/social/sql/SocialUtil.java

```diff
@@ -44,6 +47,20 @@ public static String getTenantDomain() {
 
 	}
 
+	public static String getTenantDomainFromTenantId(int tenantId) throws SocialActivityException{
+		String tenantDomainName = null;
+		try {
+			UserRealmService realmService = (UserRealmService) PrivilegedCarbonContext.getThreadLocalCarbonContext().
+					getOSGiService(UserRealmService.class);
+			org.wso2.carbon.user.api.TenantManager tenantManager = realmService.getTenantManager();
+			tenantDomainName = tenantManager.getTenant(tenantId).getDomain();
+		} catch (UserStoreException e) {
+			log.error("Error occurred while retrieving the tenant from tenant id : " + tenantId);
```

Comment:
OK, Then what is the reason for logging the stacktrace in other methods above ? Whether they aren't api ?